src/components/commons/button/Button.styled.ts

```diff
@@ -27,37 +27,49 @@ const height = {
 export const DefaultBtn = styled.button<DefaultBtnPropTypes>`
   ${Generators.flexGenerator("row", "center", "center")};
   ${({ $size }) => {
-    switch ($size) {
-      case "xlarge":
-        return css`
-          ${({ theme }) => theme.fonts["body1-normal-semi"]};
-        `;
-      case "large":
-        return css`
-          ${({ theme }) => theme.fonts["body2-normal-semi"]};
-        `;
-      case "medium":
-        return css`
-          ${({ theme }) => theme.fonts["body2-normal-semi"]};
-        `;
-      case "small":
-        return css`
-          ${({ theme }) => theme.fonts["body2-normal-semi"]};
-        `;
-      case "xsmall":
-        return css`
-          ${({ theme }) => theme.fonts["caption1-semi"]};
-        `;
+    if (typeof $size === "string") {
+      switch ($size) {
+        case "xlarge":
+          return css`
+            width: 32.7rem;
+            height: 5.6rem;
+            ${({ theme }) => theme.fonts["body1-normal-semi"]};
+          `;
+        case "large":
+          return css`
+            width: 27.9rem;
+            height: 4.6rem;
+            ${({ theme }) => theme.fonts["body2-normal-semi"]};
+          `;
+        case "medium":
+          return css`
+            width: 15.8rem;
+            height: 4.6rem;
+            ${({ theme }) => theme.fonts["body2-normal-semi"]};
+          `;
+        case "small":
+          return css`
+            width: 13.6rem;
+            height: 4.6rem;
+            ${({ theme }) => theme.fonts["body2-normal-semi"]};
+          `;
+        case "xsmall":
+          return css`
+            width: 10.3rem;
+            height: 3.6rem;
+            ${({ theme }) => theme.fonts["caption1-semi"]};
+          `;
+      }
```

Comment:
p3) 여기까지 작성된 if문을 아래 코드로 변경하는 것은 어떤가용??
계속 중복되는 폰트들도 많고, theme도 반복해서 가져오고, width를 직접 하나하나 작성하는 것이 비효율적인 것 같아서 어차피 js 문법을 사용할거면 font를 따로 변수로 만들어서 정의하고 한번에 return하는 것이 깔끔할 것 같습니다!!
```suggestion
     ${({ $size, theme }) => {
    if (typeof $size === "string") {
      let font;
      switch ($size) {
        case "xlarge":
          font = theme.fonts["body1-normal-semi"];
          break;
        case "large":
        case "medium":
        case "small":
          font = theme.fonts["body2-normal-semi"];
          break;
        case "xsmall":
          font = theme.fonts["caption1-semi"];
          break;
        default:
          font = theme.fonts["body2-normal-semi"];
      }
      return css`
        ${font};
        width: ${width[$size]};
        height: ${height[$size]};
      `;
    } 
```

Reply:
최고다 김채현

Reply:
헉 그렇네요........!! 반영하겠습니다 레전드띠예